### PR TITLE
Fix required in value schema

### PR DESCRIPTION
**Pull request by sacheu**

### CODE REVIEW

File: src/AutorestSwift/View Models/OperationViewModel.swift [hunk 1]
```diff
@@ -160,6 +160,7 @@ struct OperationViewModel {
         )
 
         for param in optionsParams {
+            // let required = param.required ?? false
```
Comment (tjprescott) [inline src/AutorestSwift/View Models/OperationViewModel.swift:163]: You can delete this.

File: src/AutorestSwift/View Models/RequestViewModel.swift [hunk 2]
```diff
@@ -78,6 +78,6 @@ private func getBodyParameters(signatureParameters: [Parameter]?) -> [Parameter]
 private func belongsInSignature(param: Parameter?) -> Bool {
     guard let httpParam = param?.protocol.http as? HttpParameter else { return false }
 
-    let required = [ParameterLocation.path, ParameterLocation.uri]
+    let required = [ParameterLocation.path, ParameterLocation.uri, ParameterLocation.body]
```
Comment (tjprescott) [inline src/AutorestSwift/View Models/RequestViewModel.swift:81]:
```suggestion
    let required: [ParameterLocation] = [.path, .uri, .body]
```

File: src/AutorestSwift/View Models/OperationViewModel.swift (hunk 1 above)
Comment (sacheu) [reply src/AutorestSwift/View Models/OperationViewModel.swift]: Removed.

File: src/AutorestSwift/View Models/KeyValueViewModel.swift [hunk 3]
```diff
@@ -61,7 +61,7 @@ struct KeyValueViewModel {
         } else if let signatureParameter = operation.signatureParameter(for: param.name) {
             // value is referring a signautre parameter, no need to wrap as String
             self.paramName = param.name
-            self.optional = signatureParameter.required ?? true
+            self.optional = !signatureParameter.required
```
Comment (tjprescott) [inline src/AutorestSwift/View Models/KeyValueViewModel.swift:64]: What is the `!` for here? I've not seen that syntax before, but it seems like a "crash if nil" thing that we should avoid.

File: src/AutorestSwift/View Models/RequestViewModel.swift (hunk 2 above)
Comment (sacheu) [reply src/AutorestSwift/View Models/RequestViewModel.swift]: Changed.

**Pull request merged**